doc/shared_library_howto/howto.txt

```diff
@@ -0,0 +1,107 @@
+
+Compiling and Using Theano functions as shared libraries
+========================================================
+
+First one important note: this functionality is experimental, so not every
+features works, and some may not work as you expected.
+
+This functionality was only be tested on CPU code on a Linux machine.
+
+For now, shared variables not works.
+
+Clearly every node of function that you want to use as shared library must
+have a C implementation.
+ 
+To compile a function as shared library, you must use the CLinker with the the
+parameter c_callable=True:
+
+linker = theano.gof.CLinker(c_callable=True)
+mode = theano.Mode(linker=linker)
+f = theano.function(inputs, outputs , mode=mode)
+
+The shared library is build in a specific temp directory, in the Theano cache
+directory. To know in which directory, you can call print the fn.name
+property of the function
```

Comment:
Rewrite the second sentence as:

The name of the directory is stored in the name property of the function.

Also since the example below disagrees which one is right?  Is it `name` or `filename`